control/ni/devices.py

```diff
@@ -0,0 +1,212 @@
+        wavelength = 1/self.freq  # seconds
+        n_waves = frame_time/wavelength
+        points_per_wave = int(np.ceil(n_points/n_waves))
+        up = np.linspace(-1, 1, points_per_wave//2 + 1)
+        down = np.linspace(1, -1, points_per_wave//2 + 1)
+        frame = np.hstack((down[:-1], np.tile(np.hstack((up[:-1], down[:-1])), round(n_waves + 20)), up[:-1]))
+        frame = ndimage.gaussian_filter1d(frame, points_per_wave/20)
+        frame = frame[points_per_wave//2:n_points + points_per_wave//2]
+        frame = frame*(self.amp/frame.max()) + self.offset
+        frame = np.hstack([np.ones(settings.readout_points//2)*frame[-1],
+                           frame,
+                           np.ones(settings.readout_points//2)*frame[-1]])
+        # frame = self.add_delays(frame, settings)
+        # frame = np.expand_dims(frame, 0)
+        return frame
+
+    def plot(self):
+        daq_data = self.one_frame(self.settings)
+        x = np.divide(list(range(daq_data.shape[1])), self.settings.final_sample_rate/1000)
+        plt.step(x, daq_data[0, :])
+
+
+# class LED(DAQDevice):
+    # def __init__(self, settings:NIDAQSettings = NIDAQSettings()):
+    #     self.settings = settings
+    #     self.blank_voltage = 10
+    #     self.power = 2
+    #     self.speed_adjustment = 0.98
+    #     self.adjusted_readout = self.settings.camera_readout_time * self.speed_adjustment
+
+    # def one_frame(self, settings: Union[NIDAQSettings, None] = None, power = None) -> np.ndarray:
+    #     if power is not None:
+    #         self.power = power
+    #     if settings is not None:
+    #         self.settings = settings
+    #         self.adjusted_readout = self.settings.camera_readout_time * self.speed_adjustment
+    #     n_shift = (round(self.settings.camera_readout_time * self.settings.final_sample_rate) -
+    #                round(self.settings.final_sample_rate * self.adjusted_readout))
+    #     n_points = (self.settings.n_points -
+    #                 round(self.adjusted_readout * self.settings.final_sample_rate))
+    #     led = np.ones(n_points) * self.power
+    #     led = np.expand_dims(led, 0)
+    #     led = self.add_readout(led)
+    #     return led
+
+    # def add_readout(self, frame:np.ndarray) -> np.ndarray:
+    #     n_shift = (round(self.settings.camera_readout_time * self.settings.final_sample_rate) -
+    #                round(self.settings.final_sample_rate * self.adjusted_readout))
+    #     print("LED readout")
+    #     print(n_shift)
+    #     readout_delay0 = np.zeros((frame.shape[0],
+    #                                round(self.settings.final_sample_rate * self.settings.camera_readout_time) - n_shift))
+    #     readout_delay1 = np.zeros((frame.shape[0],
+    #                                round(self.settings.final_sample_rate * self.settings.camera_readout_time)))
+    #     frame = np.hstack([readout_delay0, frame, readout_delay1])
+    #     return frame
```

Comment:
Can we either delete these comments or uncomment them?

Reply:
The LED is not yet adapted from the original version. Will do that tomorrow and uncomment.